src/python/epicscorelibs/test/test_ioc_ca.py

```diff
@@ -0,0 +1,97 @@
+import subprocess
+import os
+import sys
+import ctypes
+import tempfile
+import random
+import string
+
+try:
+    from queue import Queue
+except ImportError:
+    from Queue import Queue
+
+from epicscorelibs.ca import cadef, dbr
+
+
+class Channel:
+    def __init__(self, name):
+        self.name = name
+        self.chid = ctypes.c_void_p()
+
+    @property
+    def _as_parameter_(self):
+        # Setting this allows a channel object to autoconvert into the chid
+        # when passed to ca_ functions.
+        return self.chid.value
+
+
+def test_running_ioc():
+    if os.environ.get("PYTAG", "") == "win_amd64" \
+            and sys.version_info[:2] == (3, 4):
+        # This particular combination is broken. All the callbacks seem to
+        # return junk. Not sure why
+        return
+
+    # Get something reasonably unique for the PV prefix
+    pv = ''.join(random.choice(string.ascii_uppercase) for _ in range(12))
+    dbfile = tempfile.NamedTemporaryFile(delete=False)
+    dbfile.write(("""
+record(stringin, "%s") {
+    field("VAL", "blah")
+}
+""" % pv).encode())
+    dbfile.close()
+    proc = subprocess.Popen(
+        [sys.executable, '-m', 'epicscorelibs.ioc', '-d', dbfile.name],
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    )
+
+    def assert_startswith(expected):
+        l = proc.stdout.readline().decode()
+        assert l.startswith(expected), l
+
+    try:
+        # Wait for IOC to be up
+        assert_startswith("Starting iocInit")
+        assert_startswith("iocRun: All initialization complete")
```

Comment:
Just a note.  This strikes me as a bit brittle wrt. additional output.  It is also a blocking call with no timeout.  I'm not asking for a change at this time as doing this the "right" way would be more complex than is justified in test code. Actually, I find myself wondering where this wait is necessary given that the CA operations also wait.

Reply:
It isn't actually necessary, this is a leftover from when we were trying to debug windows issues. I'll remove the wait